File: 20220913/ompp-0.8.5/opari/tool/Test/test2.f90
       program test2
       integer i

       !$omp parallel
       write(*,*) "parallel"
         !$omp do
         do i=1,4
           write(*,*) "do nowait",i
         enddo
         !$omp enddo nowait

         !$omp barrier

         !$omp do
         do i=1,4
           write(*,*) "do",i
         enddo
         !$omp end do

         !$omp sections
           !$omp section
           write(*,*) "section nowait 1"
           !$omp section
           write(*,*) "section nowait 2"
         !$omp end sections nowait

         !$omp master
         write(*,*) "master"
         !$omp end master

         !$omp critical
         write(*,*) "critical"
         !$omp end critical

         !$omp critical(foobar)
         write(*,*) "critical"
         !$omp end critical(foobar)

         !$omp atomic
         ! do this atomic
         i = i + 1

         !$omp single
         write(*,*) "single"
         !$omp end single

         !$omp sections
           !$omp section
           write(*,*) "section 1"
           !$omp section
           write(*,*) "section 2"
         !$omp endsections

       !$omp end parallel

       write(*,*) "sequential1"

       !$omp parallel do
       do i=1,4
         write(*,*) "pdo",i
       enddo
       !$omp end parallel do

       write(*,*) "sequential2"

       !$omp parallelsections
         !$omp section
         write(*,*) "psection 1"
         !$omp section
         write(*,*) "psection 2"
       !$omp end parallelsections

       end

File: 20220913/ompp-0.8.5/opari/tool/Test/test2.c
#include <stdio.h>
#ifdef _OPENMP
  #include <omp.h>
#endif

int main() {
  int i;

 #pragma omp parallel
 {
   printf("parallel\n");
   #pragma omp for nowait
   for(i=0; i<4; ++i) {
     printf("do nowait %d\n", i);
   }

   #pragma omp barrier

   #pragma omp for
   for(i=0; i<4; ++i) {
     printf("do %d\n", i);
   }

   #pragma omp sections nowait
   {
     #pragma omp section
     printf("section nowait 1\n");
     #pragma omp section
     { printf("section nowait 2\n"); }
   }

   #pragma omp master
   {
     printf("master\n");
   }

   #pragma omp critical
   {
     printf("critical\n");
   }

   #pragma omp critical(foobar)
   {
     printf("critical\n");
   }

   #pragma omp atomic
   /* -------------- */
   /* do this atomic */
   i += 1;
   /* -------------- */

   #pragma omp single
   {
     printf("single\n");
   }

   #pragma omp sections
   {
     #pragma omp section
     { printf("section 1\n"); }
     #pragma omp section
     printf("section 2\n");
   }
 }

 printf("sequential1\n");

 #pragma omp parallel for
 for(i=0; i<4; ++i) {
   printf("pdo %d\n", i);
 }

 printf("sequential2\n");

 #pragma omp parallel sections
 {
   #pragma omp section
   printf("psection 1\n");
   #pragma omp section
   printf("psection 2\n");
 }
}
